sema/commons/glob/globbery.py

```
@@ -0,0 +1,125 @@
+    and none of the globs in `excludes`.
+    @param root: The root path to search under.
+    @param includes: A list of globs to include.
+    @param excludes: A list of globs to exclude.
+    @param onlyFiles: If True, only return files, not directories.
+    @return: A list of paths that match the globs.
+    """
+    found: Set[Path] = set()
+    for include in includes:
+        for path in root.glob(include):
+            if any(path.match(exclude) for exclude in excludes):
+                log.debug(f"excluding {path} by {excludes}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)